akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -496,11 +497,22 @@ private final case class SavedIslandData(islandGlobalOffset: Int, lastVisitedOff
       }
     }
 
-    islandTracking.getCurrentPhase.onIslandReady()
-    islandTracking.allNestedIslandsReady()
+    def shutdownWhileMaterializingFailure =
+      new IllegalStateException("Materializer shutdown while materializing stream")
+    try {
+      islandTracking.getCurrentPhase.onIslandReady()
+      islandTracking.allNestedIslandsReady()
+
+      if (Debug) println("--- Finished materialization")
+      matValueStack.peekLast().asInstanceOf[Mat]
+
+    } catch {
+      case ex: IllegalStateException if isShutdown && ex.getMessage == "cannot create children while terminating or terminated" ⇒
+        throw shutdownWhileMaterializingFailure
```

Comment:
is this `catch` needed? looks like the `finally` would be enough

Reply:
I wanted to give a nicer error message than that "cannot create children while terminating or terminated"

Reply:
yes, but you will only throw that `shutdownWhileMaterializingFailure` when `isShutdown` is `true`, and that would be handled by the `finally` also. Also, the exception you throw inside the catch will be replaced by the exception you throw in the finally anyway.

```
scala> try throw new Exception("e1") catch {case e: Exception => throw new Exception("e2") } finally throw new Exception("e3")
java.lang.Exception: e3
```

Reply:
Ok, catch-silence and check+throw in finally it is then

Reply:
I managed to misunderstand a second time, I'll blame it on mondays.